setupext/build_ext.py

```diff
@@ -203,40 +191,41 @@ def initialize_options(self, *args):
             '-Wimplicit-function-declaration': '',
         }
         tracing = self.distribution.enable_tracing
-        if tracing:
-            replacement['-O3'] = '-O0'
+
+        # Arguments to remove so we set debugging and optimization level
+        remove_args = ['-O0', '-O1', '-O2', '-O3', '-g']
 
         for k, v in cfg_vars.items():
             if not isinstance(v, str):
                 continue
             if not k == "OPT" and not "FLAGS" in k:
                 continue
-            for r, t in replacement.items():
-                if v.find(r) != -1:
-                    v = v.replace(r, t)
-                    cfg_vars[k] = v
+
+            args = v.split()
+            for r in remove_args:
+                args = list(filter((r).__ne__, args))
+
+            cfg_vars[k] = " ".join(args)
         super().initialize_options()
 
     def _set_cflags(self):
         # set compiler flags
         c = self.compiler.compiler_type
+        jpypeLib = self.extensions[1]
```

Comment:
this static indexing will hurt us in the future, won't it? (it already did, eventually put an assertion to match the desired name at least).

Reply:
I address this with a list comprehension.   Should be a bit more future proof.